GitUpKit/Core/GCHistory.h

```diff
@@ -95,6 +95,18 @@ typedef NS_ENUM(NSUInteger, GCHistorySorting) {
 - (BOOL)reloadHistory:(GCHistory*)history referencesDidChange:(BOOL*)referencesDidChange addedCommits:(NSArray**)addedCommits removedCommits:(NSArray**)removedCommits error:(NSError**)error;
 
 - (GCHistory*)loadHistoryFromSnapshot:(GCSnapshot*)snapshot usingSorting:(GCHistorySorting)sorting error:(NSError**)error;
+@end
+
+@interface GCRepositoryHistoryFileOptions : NSObject
+@property (nonatomic, readonly) BOOL followRenames;
+@property (nonatomic, readonly) BOOL includeMerges;
+@property (nonatomic, readonly) BOOL shouldIteroverWhenObjectNotFoundInTreeEntryByPath;
+- (instancetype)followRenames:(BOOL)followRenames;
+- (instancetype)includeMerges:(BOOL)includeMerges;
+- (instancetype)shouldIteroverWhenObjectNotFoundInTreeEntryByPath:(BOOL)shouldIteroverWhenObjectNotFoundInTreeEntryByPath;
```

Comment:
These aren't initializers...you can't treat them as such. They need to follow `init` conventions if they're to initialize an object.

Reply:
@lucasderraugh 
Do you really want all of them in one initializer? I remember the longest initializer ever in, well, 100 words?

Reply:
They don't need to be 1, but they need to follow standard Obj-C conventions of instance type instance methods that initialize objects need to start with `init` and they need to call super init. Otherwise you never initialized the object.

Reply:
`+alloc` return `instancetype`. If you invoke any property, say, `[[UIViewController alloc] view]`, you will catch crash or even undefined behavior.

Reply:
I understand that. My point is that generally speaking in Obj-C we don't return instancetype unless we're creating a new instance. Here, you aren't doing that. You're mutating the existing instance and then returning it, which is very unexpected. I can't think of anything in Foundation or AppKit that would work this way.

Reply:
@lucasderraugh Yes, we could use struct with flags, it is more appropriate for ObjC since it doesn't have convenient alternatives ( no, `NS_OPTIONS` are not convenient ). But should we?

Reply:
@lucasderraugh Or we could add NSCopying and create new objects in these methods.

Reply:
Please delete these.